### ISSUE TITLE
Unrecognized React JSX component declaration

### ISSUE THREAD

**Issue by PavelPZ** [Microsoft/TypeScript#7020]
I have this type definition file (principle extracted from https://www.nuget.org/packages/material-ui.TypeScript.DefinitelyTyped/):

```javascript
declare module TestMod {
  type TTestStyles = 'x' | 'y' | 'z';
  interface TestProps extends React.Props<TestClass> {
    myStyle: TTestStyles;
  }
  interface TestClass extends React.ComponentClass<TestProps> { }
  var Test: TestClass;
}
```

Using this definition in JSX does not work as expected:

```javascript
var Test: TestMod.TestClass;

var test = <div>
  <Test myStyle='yy'/> //Display error as expected (Type '"yy"' is not assignable to type '"x" | "y" | "z")
  <TestMod.Test myStyle='yy'/> //Does not display any error
</div>
```

**Comment by RyanCavanaugh** [comment]
See https://github.com/Microsoft/TypeScript/wiki/FAQ#i-wrote-declare-var-mycomponent-reactcomponent-why-cant-i-write-mycomponent--

**Comment by RyanCavanaugh** [comment]
Fix pending at #7030

**Comment by RyanCavanaugh** [comment]
Fixed; you should see this in the next `typescript@next` build.

Thanks for the great repro!
